contracts/vanchor/lib.rs

```diff
@@ -118,6 +212,259 @@ mod vanchor {
         }
 
         #[ink(message)]
-        pub fn do_nothing_yet(&self) {}
+        pub fn update_vanchor_config(&mut self, max_ext_amt: u128, max_fee: u128) -> Result<()> {
+            if self.creator != Self::env().caller() {
+                return Err(Error::UnknownRoot);
+            }
+
+            self.max_ext_amt = max_ext_amt;
+            self.max_fee = max_fee;
+
+            Ok(())
+        }
+
+        #[ink(message)]
+        pub fn update_edge(
+            &mut self,
+            src_chain_id: u64,
+            root: [u8; 32],
+            latest_leaf_index: u32,
+            target: [u8; 32],
+        ) -> Result<()> {
+            let edge = Edge {
+                chain_id: src_chain_id,
+                root,
+                latest_leaf_index,
+                target,
+            };
+
+            self.linkable_tree.update_edge(edge);
+
+            Ok(())
+        }
+
+        #[ink(message)]
+        pub fn transact_deposit(
+            &mut self,
+            proof_data: ProofData,
+            ext_data: ExtData,
+            recv_token_addr: AccountId,
+            recv_token_amt: u128,
+        ) -> Result<()> {
+            if self.tokenwrapper_addr != recv_token_addr {
+                return Err(Error::Unauthorized);
+            }
+
+            self.validate_proof(proof_data.clone(), ext_data.clone());
+
+            let ext_data_fee: u128 = ext_data.fee.clone();
+            let ext_amt: i128 = ext_data.ext_amount.parse().expect("Invalid ext_amount");
+            let abs_ext_amt = ext_amt.unsigned_abs();
+
+            let is_withdraw = ext_amt.is_negative();
+
+            if is_withdraw {
+                return Err(Error::InvalidExecutionEntry);
+            } else {
+                if abs_ext_amt > self.max_deposit_amt {
+                    return Err(Error::InvalidDepositAmount);
+                };
+                if abs_ext_amt != recv_token_amt {
+                    return Err(Error::InsufficientFunds);
+                };
+            }
+
+            let fee_exists = ext_data_fee != 0;
+
+            if fee_exists {
```

Comment:
If this function is exclusively for depositing then I don't see how a relayer would ever be participating or why there would be a fee. Also separately, fees are only ever taken in the wrapped token being deposited.

Reply:
For this, I actually implemented this based on the protocol-cosmwasm implementation here(https://github.com/webb-tools/protocol-cosmwasm/blob/main/contracts/vanchor/src/contract.rs#L355)
But, I know you told me earlier to look closely with the protocol-solidity implementation too. Which was what I did with the token-wrapper implementation. 
I will take a very good look at the protocol-solidity side too for this vanchor implementation and revert.